src/main/scala/edu/ie3/osmogrid/guardian/OsmoGridGuardian.scala

```diff
@@ -7,131 +7,89 @@
-          Behaviors.stopped
-        case ResultEventListenerDied =>
-          ctx.log.error("Result event listener died. That's bad...")
-          Behaviors.stopped
-        case LvCoordinatorDied =>
-          ctx.log.error("Lv coordinator died. That's bad...")
-          Behaviors.stopped
-        case unsupported =>
-          ctx.log.error(s"Received unsupported message '$unsupported'.")
-          Behaviors.stopped
-      }
-    }
+  /** Relevant, state-independent data, the the actor needs to know
+    *
+    * @param runs
+    *   Currently active conversion runs
+    */
+  private[guardian] final case class GuardianData(
+      runs: Seq[UUID]
+  ) {
+    def append(run: UUID): GuardianData = this.copy(runs = runs :+ run)
 
-  private def awaitLvGrids(
-      inputDataProvider: ActorRef[InputDataEvent],
-      resultListener: ActorRef[ResultEvent]
-  ): Behaviors.Receive[OsmoGridGuardianEvent] =
-    Behaviors.receive { (ctx, msg) =>
-      msg match {
-        case RepLvGrids(lvGrids) =>
-          ctx.log.info(s"Received ${lvGrids.length} lv grids. Join them.")
-          Try(ContainerUtils.combineToJointGrid(lvGrids.asJava)) match {
-            case Success(jointGrid) =>
-              resultListener ! GridResult(jointGrid, ctx.self)
-              awaitShutDown(inputDataProvider)
-            case Failure(exception) =>
-              ctx.log.error(
-                "Combination of received sub-grids failed. Shutting down."
-              )
-              Behaviors.stopped
-          }
-        case unsupported =>
-          ctx.log.error(
-            s"Received unsupported message while waiting for lv grids. Unsupported: $unsupported"
-          )
-          Behaviors.stopped
-      }
-    }
-
-  private def awaitShutDown(
-      inputDataProvider: ActorRef[InputDataEvent],
-      resultListenerTerminated: Boolean = false,
-      inputDataProviderTerminated: Boolean = false
-  ): Behaviors.Receive[OsmoGridGuardianEvent] = Behaviors.receive {
-    (ctx, msg) =>
-      msg match {
-        case ResultEventListenerDied =>
-          ctx.log.info("Result listener finished handling the result.")
-          ctx.log.debug("Shut down input data provider.")
-          awaitShutDown(inputDataProvider, resultListenerTerminated = true)
-        case InputDataProviderDied if resultListenerTerminated =>
-          /* That's the fine case */
-          ctx.log.info("Input data provider shut down.")
-          Behaviors.stopped
-        case InputDataProviderDied =>
-          /* That's the malicious case */
-          ctx.log.error(
-            "Input data provider unexpectedly died during shutdown was initiated."
-          )
-          Behaviors.stopped
-        case unsupported =>
-          ctx.log.error(s"Received an unsupported message $unsupported.")
-          Behaviors.stopped
-      }
+    def remove(run: UUID): GuardianData =
+      this.copy(runs = runs.filterNot(_ == run))
+  }
+  object GuardianData {
```

Comment:
Private pls (otherwise the corresponding case class would escape private visibility)

```suggestion
  private object GuardianData {
```